Sprint_6

goyya-yuri opened the pull request.

goyya-yuri committed:
Sprint_6 Test_Coverage

mkouznetsov-git commented on src/test/java/com/test/LionTest.java:
```diff
@@ -0,0 +1,53 @@
+        return Arrays.asList(new Object[][]{
+                {"Самец", true},
+                {"Самка", false},
+                {"Особь", false},
```
Некорректно ожидать значение false для кейса, в котором должно кинуться исключение.

mkouznetsov-git commented on src/test/java/com/test/LionTest.java:
```diff
@@ -0,0 +1,53 @@
+    }
+
+    @Test
+    public void getFoodTest() throws Exception {
```
Некорректный тест. Почему мы ожидаем notNull?

mkouznetsov-git commented on src/test/java/com/test/CatTest.java:
```diff
@@ -0,0 +1,27 @@
+
+    @Test
+    public void getFoodTest() throws Exception {
+        assertEquals(List.of(),cat.getFood());
```
Что проверяет этот тест? Надо убедиться, что при вызове getFood() дёргается метод feline.eatMeat()

goyya-yuri committed:
Добавлен класс для тестирования исключений

Разделение его от параметризированного класса помогает сделать код более читаемым и простым в потдержке, результаты тестирования не искажаются за счет намеренно пропущенных тестов.

Added a class for testing exceptions

Separating it from the parameterized class helps make the code more readable and easier to maintain, and the test results are not distorted by intentionally skipped tests.
